src/files/utils.ts

```diff
@@ -0,0 +1,72 @@
+import { UseMutateAsyncFunction } from '@tanstack/react-query';
+
+import { UploadSignedUrlInput } from './schemas';
+
+export const fetchFile = (url: string, metadata?: string[]) => async () => {
```

Comment:
I think jsdocs could be added on these 2 functions to explains usage (and know why   we need to return a function)